folding-schemes-solidity/src/utils/encoding.rs

```diff
@@ -0,0 +1,45 @@
+/// Defines encodings of G1 and G2 elements for use in Solidity templates.
+use ark_bn254::{Fq, G1Affine, G2Affine};
+use std::fmt::{self, Display};
+
+#[derive(Debug, Default)]
+pub struct FqWrapper(pub Fq);
+
+impl Display for FqWrapper {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self.0)
+    }
+}
+
+#[derive(Debug, Default)]
+pub struct G1Repr(pub [FqWrapper; 2]);
+
+impl Display for G1Repr {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{:#?}", self.0)
+    }
+}
+
+#[allow(dead_code)]
```

Comment:
Ah yes, this is due to the fact that clippy would otherwise complain. Since those are only used within tests, clippy marks it as dead code. Do you think there could be some better way to achieve this?

Reply:
Ah I see, because I commented it but didn't found clippy warning, but the fact is the `Groth16Verifier::from` uses them but still marked with `#[allow(dead_code)]`.

I think if we expose the structure in `verifier::template` then it wouldn't be dead code anymore, for example in `lib.rs`:

```rust
pub use verifiers::templates::*;
```

Shouldn't these be exposed for further usage?

Reply:
Yes I think we can anticipate that those will be used, notably by the upcoming cli. This also results in cleaner code, so changed all that with [cb3561d](https://github.com/privacy-scaling-explorations/folding-schemes/pull/70/commits/cb3561dcebac4f7cb9b0bd6b0af1dd567c136fe4), exposing utilities for interacting with the evm and templates. Thanks for suggesting this!

Reply:
I see, make sense